atlp-rwanda/e-commerce-furebo-32-bn
# Cart Functionalites

What does this PR do?
-------------------------
This PR enhances the functionality related to the shopping cart feature. It includes additions, updates, and improvements to the cart management system, covering operations such as adding items to the cart, viewing the cart contents, updating cart items, and clearing the cart.

Describe your changes
-------------------------
-Addition of Cart Management Services: Implemented services for adding, viewing, updating, and clearing the shopping cart, providing robust functionality for managing cart items.

-Integration with Database Models: Integrated the cart services with the database models to ensure seamless interaction between the application logic and the database.

-Endpoint Implementations: Created endpoint functions to handle HTTP requests for adding items to the cart, viewing cart contents, updating cart items, and clearing the cart.

-Test Coverage: Added comprehensive test cases to ensure the correctness and reliability of the cart management functionality, covering various scenarios and edge cases.

-Documentation Updates: Included documentation updates to reflect the changes made to the cart-related functionality, ensuring that developers have clear guidance on using and understanding the new features.

 PR Type
---------

What kind of change does this PR introduce?
[x] Feature
[ ] Bugfix
[ ] Refactoring (no functional changes, no API changes)
[ ] Documentation content changes
[ ] Tests
[ ] Other

How should this be manually tested?
----------------------------------------

-Add items to the cart using the provided endpoint and verify that the items are correctly added.

-View the cart contents through the appropriate endpoint and ensure that all added items are displayed accurately.

-Update the quantity of items in the cart and verify that the changes reflect correctly in the cart.

-Clear the cart using the designated endpoint and confirm that the cart is emptied as expected.
